src/test/java/uk/gov/hmcts/reform/sscs/service/CcdNotificationsPdfServiceTest.java

```diff
@@ -163,6 +163,50 @@ public void mergeLetterCorrespondenceIntoCcd() {
         verify(ccdService).updateCaseWithoutRetry(any(), any(), any(), eq("Notification sent"), eq("Notification sent via Gov Notify"), any());
     }
 
+    @Test
+    public void shouldMergeLetterCorrespondenceIntoCcdV2() {
+        byte[] bytes = "String".getBytes();
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
+        Correspondence correspondence = Correspondence.builder().value(
+                CorrespondenceDetails.builder()
+                        .sentOn("22 Jan 2021 11:00")
+                        .from("from")
+                        .to("to")
+                        .subject("a subject")
+                        .eventType("event")
+                        .correspondenceType(CorrespondenceType.Email)
+                        .build()).build();
+
+
+        service.mergeLetterCorrespondenceIntoCcdV2(bytes, caseId, correspondence);
+        verify(pdfStoreService).store(any(), eq("event 22 Jan 2021 11:00.pdf"), eq(CorrespondenceType.Email.name()));
+        verify(updateCcdCaseService).updateCaseV2(
+                eq(caseId), eq(EventType.NOTIFICATION_SENT.getCcdType()), eq("Notification sent"), eq("Notification sent via Gov Notify"), any(), any(Consumer.class));
```

Comment:
Can we add a Argument Captor for the Consumer here to test the logic there?

Reply:
@ruban72 added ArgumentCaptor for consumer in tests